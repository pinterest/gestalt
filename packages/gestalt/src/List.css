.bareList:first-child {
  padding-inline-start: 0;
}

.parentList {
  margin: 0;
}

.list {
  margin-left: 0;
  margin-right: 0;
  padding-inline-start: 16px;
}

.listItem {
  margin-bottom: 16px;
  margin-top: 16px;
}

/* last should precede first for single item lists */

.listItem:last-child {
  margin-bottom: 0;
  margin-top: 16px;
}

.listItem:first-child {
  margin-bottom: 16px;
  margin-top: 0;
}

.listItemCondensed {
  margin-bottom: 8px;
  margin-top: 8px;
}

/* last should precede first for single item lists */

.listItemCondensed:last-child {
  margin-bottom: 0;
  margin-top: 8px;
}

.listItemCondensed:first-child {
  margin-bottom: 8px;
  margin-top: 0;
}

/* last should precede first for single item lists */

.list .list > .listItem:last-child {
  margin-bottom: 16px;
  margin-top: 16px;
}

.list .list > .listItem:first-child {
  margin-bottom: 16px;
  margin-top: 16px;
}

.list .list > .listItemCondensed:last-child {
  margin-bottom: 8px;
  margin-top: 8px;
}

.list .list > .listItemCondensed:first-child {
  margin-bottom: 8px;
  margin-top: 8px;
}

.noStyle {
  list-style-type: none;
}

.ulItemDot {
  list-style-type: disc;
}

.ulItemCircle {
  list-style-type: circle;
}

.olItem1 {
  list-style-type: decimal;
}

.olItemA {
  list-style-type: upper-latin;
}

.olItema {
  list-style-type: lower-latin;
}

.textColorOverrideLight * {
  /* stylelint-disable-next-line declaration-no-important */
  color: var(--color-text-light) !important;
}

.textColorOverrideDark * {
  /* stylelint-disable-next-line declaration-no-important */
  color: var(--color-text-dark) !important;
}
