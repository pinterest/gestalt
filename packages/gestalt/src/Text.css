/*
This file was generated by running

  $ /Users/chrislloyd/.nvm/versions/node/v9.2.1/bin/node /Users/chrislloyd/src/gestalt/scripts/text.js

*/

:root {
  --font-size-1: 12px;
  --font-size-2: 14px;
  --font-size-3: 16px;
  --font-size-4: 18px;
  --font-size-5: 21px;
}

.Text {
  composes: antialiased sansSerif letterSpacing from "./Typography.css";
}

.fontSize1 {
  font-size: var(--font-size-1);
}

.fontSize2 {
  font-size: var(--font-size-2);
}

.fontSize3 {
  font-size: var(--font-size-3);
}

.fontSize4 {
  font-size: var(--font-size-4);
}

.fontSize5 {
  font-size: var(--font-size-5);
}

@media (--sm) {
  .smFontSize1 {
    font-size: var(--font-size-1);
  }

  .smFontSize2 {
    font-size: var(--font-size-2);
  }

  .smFontSize3 {
    font-size: var(--font-size-3);
  }

  .smFontSize4 {
    font-size: var(--font-size-4);
  }

  .smFontSize5 {
    font-size: var(--font-size-5);
  }
}

@media (--md) {
  .mdFontSize1 {
    font-size: var(--font-size-1);
  }

  .mdFontSize2 {
    font-size: var(--font-size-2);
  }

  .mdFontSize3 {
    font-size: var(--font-size-3);
  }

  .mdFontSize4 {
    font-size: var(--font-size-4);
  }

  .mdFontSize5 {
    font-size: var(--font-size-5);
  }
}

@media (--lg) {
  .lgFontSize1 {
    font-size: var(--font-size-1);
  }

  .lgFontSize2 {
    font-size: var(--font-size-2);
  }

  .lgFontSize3 {
    font-size: var(--font-size-3);
  }

  .lgFontSize4 {
    font-size: var(--font-size-4);
  }

  .lgFontSize5 {
    font-size: var(--font-size-5);
  }
}
