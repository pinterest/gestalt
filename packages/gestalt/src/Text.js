// @flow

import * as React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import colors from './Colors.css';
import styles from './Text.css';
import typography from './Typography.css';

const SIZE_SCALE: { [size: ?string]: number } = {
  xs: 1,
  sm: 2,
  md: 3,
  lg: 4,
  xl: 5,
};

type Props = {|
  align?: 'left' | 'right' | 'center' | 'justify',
  bold?: boolean,
  children?: React.Node,
  color?:
    | 'green'
    | 'pine'
    | 'olive'
    | 'blue'
    | 'navy'
    | 'midnight'
    | 'purple'
    | 'orchid'
    | 'eggplant'
    | 'maroon'
    | 'watermelon'
    | 'orange'
    | 'darkGray'
    | 'gray'
    | 'lightGray'
    | 'red'
    | 'white',
  inline?: boolean,
  italic?: boolean,
  overflow?: 'normal' | 'breakWord',
  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl',
  smSize?: 'xs' | 'sm' | 'md' | 'lg' | 'xl',
  mdSize?: 'xs' | 'sm' | 'md' | 'lg' | 'xl',
  lgSize?: 'xs' | 'sm' | 'md' | 'lg' | 'xl',
  leading?: 'tall' | 'short',
  truncate?: boolean,
  __dangerouslyIncreaseLineHeight?: boolean,
|};

export default function Text({
  align = 'left',
  bold = false,
  children,
  color = 'darkGray',
  inline = false,
  italic = false,
  overflow = 'breakWord',
  size = 'md',
  smSize,
  mdSize,
  lgSize,
  leading,
  truncate = false,
  __dangerouslyIncreaseLineHeight = false,
}: Props) {
  const scale = SIZE_SCALE[size];
  const smScale = SIZE_SCALE[smSize];
  const mdScale = SIZE_SCALE[mdSize];
  const lgScale = SIZE_SCALE[lgSize];

  const cs = cx(
    styles.Text,
    styles[`fontSize${scale}`],
    lgSize && styles[`lgFontSize${lgScale}`],
    mdSize && styles[`mdFontSize${mdScale}`],
    smSize && styles[`smFontSize${smScale}`],
    color === 'blue' && colors.blue,
    color === 'darkGray' && colors.darkGray,
    color === 'eggplant' && colors.eggplant,
    color === 'gray' && colors.gray,
    color === 'green' && colors.green,
    color === 'lightGray' && colors.lightGray,
    color === 'maroon' && colors.maroon,
    color === 'midnight' && colors.midnight,
    color === 'navy' && colors.navy,
    color === 'olive' && colors.olive,
    color === 'orange' && colors.orange,
    color === 'orchid' && colors.orchid,
    color === 'pine' && colors.pine,
    color === 'purple' && colors.purple,
    color === 'red' && colors.red,
    color === 'watermelon' && colors.watermelon,
    color === 'white' && colors.white,
    leading === 'short' && typography.leadingShort,
    (leading === 'tall' || __dangerouslyIncreaseLineHeight) &&
      typography.leadingTall,
    align === 'center' && typography.alignCenter,
    align === 'justify' && typography.alignJustify,
    align === 'left' && typography.alignLeft,
    align === 'right' && typography.alignRight,
    overflow === 'breakWord' && typography.breakWord,
    italic && typography.fontStyleItalic,
    !italic && typography.fontStyleNormal,
    bold && typography.fontWeightBold,
    !bold && typography.fontWeightNormal,
    truncate && typography.truncate
  );
  const Tag = inline ? 'span' : 'div';

  return (
    <Tag
      className={cs}
      {...(truncate && typeof children === 'string'
        ? { title: children }
        : null)}
    >
      {children}
    </Tag>
  );
}

Text.propTypes = {
  __dangerouslyIncreaseLineHeight: PropTypes.bool,
  align: PropTypes.oneOf(['left', 'right', 'center', 'justify']),
  bold: PropTypes.bool,
  children: PropTypes.node,
  color: PropTypes.oneOf([
    'green',
    'pine',
    'olive',
    'blue',
    'navy',
    'midnight',
    'purple',
    'orchid',
    'eggplant',
    'maroon',
    'watermelon',
    'orange',
    'darkGray',
    'gray',
    'lightGray',
    'red',
    'white',
  ]),
  inline: PropTypes.bool,
  italic: PropTypes.bool,
  leading: PropTypes.oneOf(['tall', 'short']),
  lgSize: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl']),
  mdSize: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl']),
  overflow: PropTypes.oneOf(['normal', 'breakWord']),
  size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl']),
  smSize: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl']),
  truncate: PropTypes.bool,
};
