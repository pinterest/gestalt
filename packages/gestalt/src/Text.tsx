import { forwardRef, ReactElement, ReactNode } from 'react';
import cx from 'classnames';
import styles from './Text.css';
import { semanticColors } from './textTypes';
import typography from './Typography.css';

function isNotNullish(val?: number | null): boolean {
  return val !== null && val !== undefined;
}

type As = 'span' | 'div';
type Overflow = 'normal' | 'breakAll' | 'breakWord' | 'noWrap';
type Size = '100' | '200' | '300' | '400' | '500' | '600';

type Props = {
  /**
   * `"start"` and `"end"` should be used for regular alignment since they flip with locale direction. `"forceLeft"` and `"forceRight"` should only be used in special cases where locale direction should be ignored, such as tabular or numeric text. See the [alignment variant](https://gestalt.pinterest.systems/web/text#Alignment) for more details.
   */
  align?: 'start' | 'end' | 'center' | 'forceLeft' | 'forceRight';
  /**
   * The text content to be displayed.
   */
  children?: ReactNode;
  /**
   * The color of the text content. See the [colors variant](https://gestalt.pinterest.systems/web/text#Colors) for more details.
   */
  color?:
    | 'default'
    | 'disabled'
    | 'subtle'
    | 'success'
    | 'error'
    | 'warning'
    | 'shopping'
    | 'link'
    | 'inverse'
    | 'light'
    | 'dark';
  /**
   * Indicates how the text should flow with the surrounding content. See the [block vs inline variant](https://gestalt.pinterest.systems/web/text#Block-vs.-inline) for more details.
   */
  inline?: boolean;
  /**
   * Indicates if the text should be displayed in italic. See the [styles variant](https://gestalt.pinterest.systems/web/text#Styles) for more details.
   */
  italic?: boolean;
  /**
   * Visually truncate the text to the specified number of lines. This also adds the `title` attribute if `children` is a string, which displays the full text on hover in most browsers. See the [overflow and truncation variant](https://gestalt.pinterest.systems/web/text#Overflow-and-truncation) for more details.
   */
  lineClamp?: number;
  /**
   * Indicates how the text content should handle overflowing its container. See the [overflow and truncation variant](https://gestalt.pinterest.systems/web/text#Overflow-and-truncation) for more details.
   */
  overflow?: Overflow;
  /**
   * Ref that is forwarded to the underlying element. See the [ref variant](https://gestalt.pinterest.systems/web/text#Refs) for more details.
   */
  ref?: HTMLDivElement | HTMLSpanElement;
  /**
   * The sizes are based on our [font-size design tokens](https://gestalt.pinterest.systems/foundations/design_tokens/overview#Font-size). See the [sizes variant](https://gestalt.pinterest.systems/web/text#Sizes) for more details.
   */
  size?: Size;
  /**
   * This populates the `title` attribute of the element, which is visible on hover in most browsers. This is useful when truncating the text with `lineClamp` when `children` is a `React.Node`. See the [Title variant](https://gestalt.pinterest.systems/web/text#Title) for more details.
   */
  title?: string;
  /**
   * Indicates if the text content should be underlined. See the [styles variant](https://gestalt.pinterest.systems/web/text#Styles) for more details.
   */
  underline?: boolean;
  /**
   * Indicates the font weight for the text content. See the [styles variant](https://gestalt.pinterest.systems/web/text#Styles) for more details.
   */
  weight?: 'bold' | 'normal';
};

/**
 * [Text](https://gestalt.pinterest.systems/web/text) component is used for all non-heading text on all surfaces, whether inside of UI components or in long-form paragraph text.
 *
 * ![Text light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/Text.spec.ts-snapshots/Text-chromium-darwin.png)
 * ![Text dark mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/Text-dark.spec.ts-snapshots/Text-dark-chromium-darwin.png)
 */
const TextWithForwardRef = forwardRef<HTMLElement, Props>(function Text(
  {
    align = 'start',
    children,
    color = 'default',
    inline = false,
    italic = false,
    lineClamp,
    overflow = 'breakWord',
    size = '300',
    title,
    underline = false,
    weight = 'normal',
  }: Props,
  ref,
): ReactElement {
  const colorClass = semanticColors.includes(color) && styles[color];

  const getWordBreakStyle = (): string | undefined => {
    if (overflow === 'breakAll') {
      return typography.breakAll;
    }

    // default to breakWord if lineClamp is set
    if (overflow === 'breakWord' || isNotNullish(lineClamp)) {
      return typography.breakWord;
    }

    return undefined;
  };

  const cs = cx(
    styles.Text,
    typography[`fontSize${size}`],
    color && colorClass,
    align === 'center' && typography.alignCenter,
    // @ts-expect-error - TS2367 - This condition will always return 'false' since the types '"center" | "start" | "end" | "forceLeft" | "forceRight"' and '"justify"' have no overlap.
    align === 'justify' && typography.alignJustify,
    align === 'start' && typography.alignStart,
    align === 'end' && typography.alignEnd,
    align === 'forceLeft' && typography.alignForceLeft,
    align === 'forceRight' && typography.alignForceRight,
    getWordBreakStyle(),
    overflow === 'noWrap' && typography.noWrap,
    italic && typography.fontStyleItalic,
    underline && typography.underline,
    weight === 'bold' && typography.fontWeightSemiBold,
    weight === 'normal' && typography.fontWeightNormal,
    isNotNullish(lineClamp) && typography.lineClamp,
  );

  const Tag: As = inline ? 'span' : 'div';

  return (
    <Tag
      className={cs}
      title={
        title ?? (isNotNullish(lineClamp) && typeof children === 'string' ? children : undefined)
      }
      {...(lineClamp ? { style: { WebkitLineClamp: lineClamp } } : {})}
      // @ts-expect-error - TS2322 - Type 'ForwardedRef<HTMLElement>' is not assignable to type 'LegacyRef<HTMLDivElement> | undefined'.
      ref={ref}
    >
      {children}
    </Tag>
  );
});

TextWithForwardRef.displayName = 'Text';

export default TextWithForwardRef;
