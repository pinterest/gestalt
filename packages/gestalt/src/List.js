// @flow strict
import { type Element, type Node } from 'react';
import InternalList from './List/InternalList.js';
import ListItem from './ListItem.js'; // eslint-disable import/no-cycle
import Text from './Text.js';

type ListType = 'bare' | 'ordered' | 'unordered';

type Props = {
  /**
   * Use List.Item to build lists. See [subcomponents](https://gestalt.pinterest.systems/web/list#List.Item).
   */
  children: Node,
  /**
   * The label for the list. Be sure to localize the text. See the [label variant](https://gestalt.pinterest.systems/web/list#Text-and-label) for guidance.
   */
  label?: string | Element<typeof Text>,
  /**
   * Whether the label should be visible or not. If `hidden`, the label is still available for screen reader users, but does not appear visually. See the [accessibility section](https://gestalt.pinterest.systems/web/list#Accessibility) and the [label variant](https://gestalt.pinterest.systems/web/list#Text-and-label) for guidance.
   */
  labelDisplay?: 'visible' | 'hidden',
  /**
   * The spacing used in List. See the [spacing variant](https://gestalt.pinterest.systems/web/list#Spacing) for guidance.
   */
  spacing?: 'regular' | 'condensed',
  /**
   * Determines the style of the list. See the [type variant](https://gestalt.pinterest.systems/web/list#Type) to learn more.
   */
  type?: ListType,
};

/**
 * [List](https://gestalt.pinterest.systems/web/list) allows users to view individual, but related, text items grouped together.
 * ![List light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/List.spec.mjs-snapshots/List-chromium-darwin.png)
 * ![List dark mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/List-dark.spec.mjs-snapshots/List-dark-chromium-darwin.png)
 */
function List({
  label,
  labelDisplay = 'visible',
  spacing = 'regular',
  type,
  children,
}: Props): Node {
  return (
    // We need this InternalList to avoid the circular dependency src/List.js -> src/ListItem.js -> src/List.js
    <InternalList type={type} spacing={spacing} label={label} labelDisplay={labelDisplay}>
      {children}
    </InternalList>
  );
}

List.displayName = 'List';

List.Item = ListItem;

export default List;
