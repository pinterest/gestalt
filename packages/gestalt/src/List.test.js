// @flow strict
import { create } from 'react-test-renderer';
import List from './List';
import Text from './Text';

describe('List', () => {
  it('renders a bare unordered list', () => {
    const tree = create(
      <List label="label test" type="bare">
        <List.Item text="List item text">
          <List.Item text="List item text">
            <List.Item text="List item text">
              <List.Item text="List item text">
                <List.Item text="List item text">
                  <List.Item text="List item text" />
                </List.Item>
              </List.Item>
            </List.Item>
          </List.Item>
        </List.Item>
      </List>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders an unordered list', () => {
    const tree = create(
      <List label="label test" type="unordered">
        <List.Item text="List item text">
          <List.Item text="List item text">
            <List.Item text="List item text">
              <List.Item text="List item text">
                <List.Item text="List item text">
                  <List.Item text="List item text" />
                </List.Item>
              </List.Item>
            </List.Item>
          </List.Item>
        </List.Item>
      </List>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders an ordered list', () => {
    const tree = create(
      <List label="label test" type="ordered">
        <List.Item text="List item text">
          <List.Item text="List item text">
            <List.Item text="List item text">
              <List.Item text="List item text">
                <List.Item text="List item text">
                  <List.Item text="List item text" />
                </List.Item>
              </List.Item>
            </List.Item>
          </List.Item>
        </List.Item>
      </List>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders an mixed list', () => {
    const tree = create(
      <List label="label test" type="ordered">
        <List.Item text="List item text" />
        <List.Item text="List item text">
          <List type="unordered">
            <List.Item text="List item text" />
            <List.Item text="List item text">
              <List.Item text="List item text" />
              <List.Item text="List item text">
                <List type="ordered">
                  <List.Item text="List item text" />
                  <List.Item text="List item text" />
                </List>
              </List.Item>
            </List.Item>
            <List.Item text="List item text" />
          </List>
        </List.Item>
        <List.Item text="List item text" />
      </List>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders an list with a custom label', () => {
    const tree = create(
      <List label={<Text weight="bold">label test</Text>} type="ordered">
        <List.Item text="List item text" />
      </List>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders an list with a hidden label', () => {
    const tree = create(
      <List label="label test" labelDisplay="hidden" type="ordered">
        <List.Item text="List item text" />
      </List>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders an condensed list', () => {
    const tree = create(
      <List label="label test" spacing="condensed" type="ordered">
        <List.Item text="List item text" />
      </List>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders an list item with custom text', () => {
    const tree = create(
      <List label="label test" spacing="condensed" type="ordered">
        <List.Item text={<Text weight="bold">List item text</Text>} />
      </List>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
