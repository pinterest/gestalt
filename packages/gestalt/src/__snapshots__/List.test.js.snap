// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders a bare unordered list 1`] = `
<div
  className="Flex rowGap0 columnGap0 xsDirectionColumn"
>
  <div
    className="box marginBottom4 xsDisplayBlock"
    id=":r0:"
  >
    <div
      className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
    >
      label test
    </div>
  </div>
  <div
    className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
  >
    <ul
      aria-labelledby=":r0:"
      className="list bareList parentList"
    >
      <li
        className="noStyle listItem"
      >
        <div
          className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
        >
          List item text
        </div>
        <ul
          aria-labelledby=":r1:"
          className="list bareList"
        >
          <li
            className="noStyle listItem"
          >
            <div
              className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
            >
              List item text
            </div>
            <ul
              aria-labelledby=":r2:"
              className="list bareList"
            >
              <li
                className="noStyle listItem"
              >
                <div
                  className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                >
                  List item text
                </div>
                <ul
                  aria-labelledby=":r3:"
                  className="list bareList"
                >
                  <li
                    className="noStyle listItem"
                  >
                    <div
                      className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                    >
                      List item text
                    </div>
                    <ul
                      aria-labelledby=":r4:"
                      className="list bareList"
                    >
                      <li
                        className="noStyle listItem"
                      >
                        <div
                          className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                        >
                          List item text
                        </div>
                        <ul
                          aria-labelledby=":r5:"
                          className="list bareList"
                        >
                          <li
                            className="noStyle listItem"
                          >
                            <div
                              className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                            >
                              List item text
                            </div>
                          </li>
                        </ul>
                      </li>
                    </ul>
                  </li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>
      </li>
    </ul>
  </div>
</div>
`;

exports[`List renders an condensed list 1`] = `
<div
  className="Flex rowGap0 columnGap0 xsDirectionColumn"
>
  <div
    className="box marginBottom4 xsDisplayBlock"
    id=":ro:"
  >
    <div
      className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
    >
      label test
    </div>
  </div>
  <div
    className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
  >
    <ol
      aria-labelledby=":ro:"
      className="list parentList"
    >
      <li
        className="listItemCondensed olItem1"
      >
        <div
          className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
        >
          List item text
        </div>
      </li>
    </ol>
  </div>
</div>
`;

exports[`List renders an list item with custom text 1`] = `
<div
  className="Flex rowGap0 columnGap0 xsDirectionColumn"
>
  <div
    className="box marginBottom4 xsDisplayBlock"
    id=":rp:"
  >
    <div
      className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
    >
      label test
    </div>
  </div>
  <div
    className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
  >
    <ol
      aria-labelledby=":rp:"
      className="list parentList"
    >
      <li
        className="listItemCondensed olItem1"
      >
        <span
          className="textColorOverrideDark"
        >
          <div
            className="Text fontSize300 defaultText alignStart breakWord fontWeightSemiBold"
          >
            List item text
          </div>
        </span>
      </li>
    </ol>
  </div>
</div>
`;

exports[`List renders an list with a custom label 1`] = `
<div
  className="Flex rowGap0 columnGap0 xsDirectionColumn"
>
  <div
    className="box marginBottom4 xsDisplayBlock"
    id=":rm:"
  >
    <span
      className="textColorOverrideDark"
    >
      <div
        className="Text fontSize300 defaultText alignStart breakWord fontWeightSemiBold"
      >
        label test
      </div>
    </span>
  </div>
  <div
    className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
  >
    <ol
      aria-labelledby=":rm:"
      className="list parentList"
    >
      <li
        className="listItem olItem1"
      >
        <div
          className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
        >
          List item text
        </div>
      </li>
    </ol>
  </div>
</div>
`;

exports[`List renders an list with a hidden label 1`] = `
<div
  className="Flex rowGap0 columnGap0 xsDirectionColumn"
>
  <div
    className="box marginBottom0 xsDisplayVisuallyHidden"
    id=":rn:"
  >
    <div
      className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
    >
      label test
    </div>
  </div>
  <div
    className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
  >
    <ol
      aria-labelledby=":rn:"
      className="list parentList"
    >
      <li
        className="listItem olItem1"
      >
        <div
          className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
        >
          List item text
        </div>
      </li>
    </ol>
  </div>
</div>
`;

exports[`List renders an mixed list 1`] = `
<div
  className="Flex rowGap0 columnGap0 xsDirectionColumn"
>
  <div
    className="box marginBottom4 xsDisplayBlock"
    id=":ri:"
  >
    <div
      className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
    >
      label test
    </div>
  </div>
  <div
    className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
  >
    <ol
      aria-labelledby=":ri:"
      className="list parentList"
    >
      <li
        className="listItem olItem1"
      >
        <div
          className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
        >
          List item text
        </div>
      </li>
      <li
        className="listItem olItem1"
      >
        <div
          className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
        >
          List item text
        </div>
        <ul
          aria-labelledby=":rj:"
          className="list"
        >
          <li
            className="listItem ulItemDot"
          >
            <div
              className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
            >
              List item text
            </div>
          </li>
          <li
            className="listItem ulItemDot"
          >
            <div
              className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
            >
              List item text
            </div>
            <ul
              aria-labelledby=":rk:"
              className="list"
            >
              <li
                className="listItem ulItemCircle"
              >
                <div
                  className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                >
                  List item text
                </div>
              </li>
              <li
                className="listItem ulItemCircle"
              >
                <div
                  className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                >
                  List item text
                </div>
                <ol
                  aria-labelledby=":rl:"
                  className="list"
                >
                  <li
                    className="listItem olItem1"
                  >
                    <div
                      className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                    >
                      List item text
                    </div>
                  </li>
                  <li
                    className="listItem olItem1"
                  >
                    <div
                      className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                    >
                      List item text
                    </div>
                  </li>
                </ol>
              </li>
            </ul>
          </li>
          <li
            className="listItem ulItemDot"
          >
            <div
              className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
            >
              List item text
            </div>
          </li>
        </ul>
      </li>
      <li
        className="listItem olItem1"
      >
        <div
          className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
        >
          List item text
        </div>
      </li>
    </ol>
  </div>
</div>
`;

exports[`List renders an ordered list 1`] = `
<div
  className="Flex rowGap0 columnGap0 xsDirectionColumn"
>
  <div
    className="box marginBottom4 xsDisplayBlock"
    id=":rc:"
  >
    <div
      className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
    >
      label test
    </div>
  </div>
  <div
    className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
  >
    <ol
      aria-labelledby=":rc:"
      className="list parentList"
    >
      <li
        className="listItem olItem1"
      >
        <div
          className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
        >
          List item text
        </div>
        <ol
          aria-labelledby=":rd:"
          className="list"
        >
          <li
            className="listItem olItemA"
          >
            <div
              className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
            >
              List item text
            </div>
            <ol
              aria-labelledby=":re:"
              className="list"
            >
              <li
                className="listItem olItema"
              >
                <div
                  className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                >
                  List item text
                </div>
                <ol
                  aria-labelledby=":rf:"
                  className="list"
                >
                  <li
                    className="listItem olItem1"
                  >
                    <div
                      className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                    >
                      List item text
                    </div>
                    <ol
                      aria-labelledby=":rg:"
                      className="list"
                    >
                      <li
                        className="listItem olItemA"
                      >
                        <div
                          className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                        >
                          List item text
                        </div>
                        <ol
                          aria-labelledby=":rh:"
                          className="list"
                        >
                          <li
                            className="listItem olItema"
                          >
                            <div
                              className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                            >
                              List item text
                            </div>
                          </li>
                        </ol>
                      </li>
                    </ol>
                  </li>
                </ol>
              </li>
            </ol>
          </li>
        </ol>
      </li>
    </ol>
  </div>
</div>
`;

exports[`List renders an unordered list 1`] = `
<div
  className="Flex rowGap0 columnGap0 xsDirectionColumn"
>
  <div
    className="box marginBottom4 xsDisplayBlock"
    id=":r6:"
  >
    <div
      className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
    >
      label test
    </div>
  </div>
  <div
    className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
  >
    <ul
      aria-labelledby=":r6:"
      className="list parentList"
    >
      <li
        className="listItem ulItemDot"
      >
        <div
          className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
        >
          List item text
        </div>
        <ul
          aria-labelledby=":r7:"
          className="list"
        >
          <li
            className="listItem ulItemCircle"
          >
            <div
              className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
            >
              List item text
            </div>
            <ul
              aria-labelledby=":r8:"
              className="list"
            >
              <li
                className="listItem ulItemDot"
              >
                <div
                  className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                >
                  List item text
                </div>
                <ul
                  aria-labelledby=":r9:"
                  className="list"
                >
                  <li
                    className="listItem ulItemCircle"
                  >
                    <div
                      className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                    >
                      List item text
                    </div>
                    <ul
                      aria-labelledby=":ra:"
                      className="list"
                    >
                      <li
                        className="listItem ulItemDot"
                      >
                        <div
                          className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                        >
                          List item text
                        </div>
                        <ul
                          aria-labelledby=":rb:"
                          className="list"
                        >
                          <li
                            className="listItem ulItemCircle"
                          >
                            <div
                              className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                            >
                              List item text
                            </div>
                          </li>
                        </ul>
                      </li>
                    </ul>
                  </li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>
      </li>
    </ul>
  </div>
</div>
`;
